### PR TITLE
Add separate client for dynamic relays (fixes #2368)

### LINKED ISSUE #2363: wired layout
I often got the following layout after login but before accepting the self signed certificate.. maybe a cache problem?
wired_layout

### LINKED ISSUE #2368: Pick next relay in case of error
When the connection to a relay fails, then syncthing should choose the next relay from the pool. Currently it just indicates that there has been an error.

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
Did some manual tests in the playground, such as kicking off two clients in parallel, first connecting,
second one getting a message about already being connected, falling back to the second address.

### CODE REVIEW

**Comment by calmh** [comment]
Too big a diff to review with a phone from the airport, I'll get back on this. It sounds sane.

**Comment by rumpelsepp** [comment]
@AudriusButkevicius: Are you sure that this commit fixes #2364?

**Comment by AudriusButkevicius** [comment]
Wrong ticket number, but you know exactly what we are talking about.

**Comment by rumpelsepp** [comment]
> Wrong ticket number, but you know exactly what we are talking about.

Maybe :dancers:

**Comment by calmh** [comment]
@st-jenkins retest this please

**Comment by calmh** [comment]
Seems resonable, I just want to run some tests first. :) Mind rebasing?

**Comment by AudriusButkevicius** [comment]
Rebased

File: lib/relay/client/dynamic.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+func (c *dynamicClient) String() string {
+	return fmt.Sprintf("DynamicClient:%p:%s@%s", c, c.URI(), c.pooladdr)
+}
+
+func (c *dynamicClient) URI() *url.URL {
+	c.mut.RLock()
+	defer c.mut.RUnlock()
+	if c.client == nil {
+		return c.pooladdr
```
Comment (calmh) [inline lib/relay/client/dynamic.go]: This makes the above String() call return something like `DynamicClient:0xc8201daa80:dynamic+https://relays.syncthing.net@dynamic+https://relays.syncthing.net` on failure - should we return "noclient" or something instead?

File: lib/relay/client/static.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+	"github.com/syncthing/syncthing/lib/relay/protocol"
+	"github.com/syncthing/syncthing/lib/sync"
+)
+
+type staticClient struct {
+	uri         *url.URL
+	invitations chan protocol.SessionInvitation
+
+	closeInvitationsOnFinish bool
+
```
Comment (calmh) [inline lib/relay/client/static.go]: This is very spacious of you :)

**Pull request merged**